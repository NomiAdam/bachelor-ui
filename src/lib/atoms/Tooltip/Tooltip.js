import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { basicTheme, lightTheme } from '../../constants/theme';

const StyledTooltipText = styled.span`
  visibility: hidden;
  width: 120px;
  background-color: ${basicTheme.blue};
  color: ${lightTheme.white};
  text-align: center;
  padding: 5px 0;
  border-radius: 6px;
  position: absolute;
  z-index: 1;
  bottom: 100%;
  left: 50%; 
  margin-bottom: 5px;
  margin-left: -60px;
  &:after {
    content: "";
    position: absolute;
    top: 100%;
    left: 50%;
    margin-left: -5px;
    border: 5px solid transparent;
    border-top-color: ${basicTheme.blue};
  }
`;

const StyledTooltip = styled.div`
  position: relative;
  display: inline-block;
  &:hover {
    ${StyledTooltipText} {
      visibility: visible;
    }
  }
`;

const Tooltip = ({ label, children }) => (
    <StyledTooltip>
        {children}
        <StyledTooltipText>{ label }</StyledTooltipText>
    </StyledTooltip>
);

Tooltip.propTypes = {
    /**
     * String definition of tooltip text
     */
    label: PropTypes.string,
    /**
     * Anything to have tooltip attached to it
     */
    children: PropTypes.any,
};

export { StyledTooltip };
export default Tooltip;
