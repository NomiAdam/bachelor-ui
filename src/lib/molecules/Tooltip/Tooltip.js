import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import DEFAULT_THEME, { resolveTheme, TYPES } from '../../utils/resolveTheme';

const StyledTooltipText = styled.span`
  ${ DEFAULT_THEME }
  visibility: hidden;
  width: 120px;
  text-align: center;
  padding: 5px 0;
  position: absolute;
  z-index: 1;
  bottom: 100%;
  left: 50%; 
  margin-bottom: 5px;
  margin-left: -60px;
  &:after {
    content: "";
    position: absolute;
    top: 100%;
    left: 50%;
    margin-left: -5px;
    border: 5px solid transparent;
    border-top-color: ${ resolveTheme(TYPES.COLOR) };
  }
`;

const StyledTooltip = styled.div`
  position: relative;
  display: inline-block;
  &:hover {
    ${ StyledTooltipText } {
      visibility: visible;
    }
  }
`;

const Tooltip = ({ label, children }) => (
    <StyledTooltip>
        {children}
        <StyledTooltipText>{ label }</StyledTooltipText>
    </StyledTooltip>
);

Tooltip.propTypes = {
    /**
     * String definition of tooltip text
     */
    label: PropTypes.string,
    /**
     * Anything to have tooltip attached to it
     */
    children: PropTypes.any,
};

export { StyledTooltip };
export default Tooltip;
