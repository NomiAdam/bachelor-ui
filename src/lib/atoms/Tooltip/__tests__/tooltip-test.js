/* eslint-disable no-undef */
import React from 'react';
import Tooltip, {StyledTooltip} from '../Tooltip';
import {shallow} from 'enzyme';
import 'jest-styled-components';

describe('Tooltip', () => {
    it('should be defined', () => {
        expect(Tooltip).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Tooltip/>)).toMatchSnapshot();
    });
    it('should be of type StyledTooltip', () => {
        expect(shallow(<Tooltip />).type()).toBe(StyledTooltip);
    });
});
