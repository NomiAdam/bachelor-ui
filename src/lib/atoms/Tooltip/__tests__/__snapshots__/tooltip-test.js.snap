// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should match the snapshot 1`] = `
<Tooltip__StyledTooltip>
  <Tooltip__StyledTooltipText />
</Tooltip__StyledTooltip>
`;
